data-workflows/plugin/metadata.py

```diff
@@ -0,0 +1,47 @@
+from collections import defaultdict
+from typing import Optional
+
+from nhcommons.models.category import get_category
+from nhcommons.utils.custom_parser import render_description
+from nhcommons.utils.github_adapter import get_github_metadata
+from nhcommons.utils.pypi_adapter import get_plugin_pypi_metadata
+
+
+def get_formatted_metadata(plugin: str, version: str) -> Optional[dict]:
+    pypi_metadata = get_plugin_pypi_metadata(plugin, version)
+    if not pypi_metadata:
+        return None
+
+    metadata = _generate_metadata(pypi_metadata)
+    return _format_metadata(metadata)
+
+
+def _format_metadata(metadata: dict) -> dict:
+    if "description" in metadata:
+        description = metadata.get("description")
+        metadata["description_text"] = render_description(description)
+    if "labels" in metadata:
+        category_version = metadata["labels"]["ontology"]
+        categories = defaultdict(list)
+        category_hierarchy = defaultdict(list)
+        for label_term in metadata["labels"]["terms"]:
```

Comment:
```suggestion
    metadata = _generate_metadata(pypi_metadata)
    try:
        return _format_metadata(metadata)
    except KeyError:
        return None


def _format_metadata(metadata: dict) -> dict:
    if "description" in metadata:
        description = metadata.get("description")
        metadata["description_text"] = render_description(description)
    if "labels" in metadata:
        category_version = metadata["labels"]["ontology"]
        categories = defaultdict(list)
        category_hierarchy = defaultdict(list)
        for label_term in metadata["labels"]["terms"]:
```

I think this will raise a `KeyError` if (for example) `"ontology"` or `"terms"` is not in `metadata["labels"]`. I'm not sure if you want to catch it here but `get_formatted_metadata` is already returning an Optional. It might be nice to log the missing key as well.

Otherwise maybe this is caught higher up somewhere?

Reply:
Instead of raising a `KeyError`, I think handling this here with a warning log would be less disruptive.